r/R/dataset.R

```diff
@@ -141,6 +141,40 @@ Dataset$create <- function(children, schema) {
 #' @export
 names.Dataset <- function(x) names(x$schema)
 
+#' @export
+dim.Dataset <- function(x) {
+
+  if (!inherits(x, "FileSystemDataset")) {
+    stop(
+      "dim() is not currently implemented for ", class(x)[1] ," Datasets. ",
+      "Call collect() first to pull data into R.",
+      call. = FALSE
+    )
+  }
+
+  if (any(!grepl("\\.parquet$", basename(x$files)))) {
```

Comment:
```suggestion
  if (!inherits(x$format, "ParquetFileFormat")) {
```

And actually with this you can probably drop the first check because non-FileSystemDatasets won't have `$format` defined, so `x$format` is `NULL`